components/homme/CMakeLists.txt

```diff
@@ -370,26 +376,38 @@ SET (CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/utils/externals/scorpio/cmak
 ADD_DEFINITIONS(-D_NO_MPI_RSEND)
 
 if(NOT BUILD_HOMME_WITHOUT_PIOLIBRARY)
-SET(NetCDF_PATH ${NETCDF_DIR})
-SET(PnetCDF_PATH ${PNETCDF_DIR})
-message("** Configuring SCORPIO with:")
-message("-- NetCDF_PATH = ${NetCDF_PATH}")
-message("-- PnetCDF_PATH = ${PnetCDF_PATH}")
-message("-- NetCDF_Fortran_PATH = ${NetCDF_Fortran_PATH}")
-message("-- NetCDF_C_PATH = ${NetCDF_C_PATH}")
-message("-- PnetCDF_Fortran_PATH = ${PnetCDF_Fortran_PATH}")
-message("-- PnetCDF_C_PATH = ${PnetCDF_C_PATH}")
-# pio needs cime/externals/genf90/genf90.pl
-if (HOMME_USE_SCORPIO)
-  # Need to use scorpio's older genf90 for now
-  SET(GENF90_PATH ${CMAKE_CURRENT_SOURCE_DIR}/utils/utils/externals/scorpio/src/genf90)
-  SET(PIO_ENABLE_TOOLS OFF CACHE BOOL "Disabling Scorpio tool build")
-  ADD_SUBDIRECTORY(utils/externals/scorpio)
-else ()
-  # The default I/O library used in "Scorpio classic"
-  ADD_SUBDIRECTORY(utils/externals/scorpio_classic)
-  SET(GENF90_PATH ${CMAKE_CURRENT_SOURCE_DIR}/utils/cime/CIME/non_py/externals/genf90)
-endif ()
+  SET(NetCDF_PATH ${NETCDF_DIR})
+  SET(PnetCDF_PATH ${PNETCDF_DIR})
+  message("-- NetCDF_PATH = ${NetCDF_PATH}")
+  message("-- PnetCDF_PATH = ${PnetCDF_PATH}")
+  message("-- NetCDF_Fortran_PATH = ${NetCDF_Fortran_PATH}")
+  message("-- NetCDF_C_PATH = ${NetCDF_C_PATH}")
+  message("-- PnetCDF_Fortran_PATH = ${PnetCDF_Fortran_PATH}")
+  message("-- PnetCDF_C_PATH = ${PnetCDF_C_PATH}")
+  if (HOMME_BUILD_SCORPIO)
+    message("** Configuring SCORPIO")
+    # pio needs cime/externals/genf90/genf90.pl
+    if (HOMME_USE_SCORPIO)
+      # Need to use scorpio's older genf90 for now
+      SET(GENF90_PATH ${CMAKE_CURRENT_SOURCE_DIR}/utils/utils/externals/scorpio/src/genf90)
+      SET(PIO_ENABLE_TOOLS OFF CACHE BOOL "Disabling Scorpio tool build")
+      ADD_SUBDIRECTORY(utils/externals/scorpio)
+    else ()
+      # The default I/O library used in "Scorpio classic"
+      ADD_SUBDIRECTORY(utils/externals/scorpio_classic)
+      SET(GENF90_PATH ${CMAKE_CURRENT_SOURCE_DIR}/utils/cime/CIME/non_py/externals/genf90)
+    endif ()
+  else ()
+    # The motivation for this block is that EAMxx is building Homme. Homme
+    # shouldn't build SCORPIO, but it does still need to find netcdf.mod to read
+    # RRM grids.
+    #   Use SCORPIO's FindNetCDF.cmake to get the Fortran netcdf.mod include
+    # directory.
+    find_package(NetCDF COMPONENTS Fortran REQUIRED)
```

Comment:
This is odd. EAMxx does require Scorpio, and indeed is wrapping scorpio libs in the pioc target (hence, HOMME_BUILD_SCORPIO=FALSE). In doing this, it should already link the netcdf libs to the pioc and piof targets. `target_link_libraries (${execLibName} piof)` should then attach the netcdf deps to homme...

Reply:
Here you mean linking. But netcdf.mod is for compiling, i.e., `#include "netcdf.mod"` in mesh_mod.F90.

Reply:
`target_link_libraries` is a misleading name. If the "linked" item is a library, it simply links. But if it is a CMake target (as pioc and piof are), it does eveyrhing: it links, but also adds PUBLIC include dirs of the target, and propagate PUBLIC flags/CPP of the target. That's basically the only reason I create the pioc/piof imported CMake targets in EAMxx, so that one only needs to do `target_link_libraries(the_tgt pioc)` to get everything they need from pioc.

Reply:
TBH, I should first check that the folder where netcdf.mod is located is included in the INTERFACE_INCLUDE_DIRECTORIES section of `piof` target...

Reply:
Ok. I think I tried that and concluded that wasn't working, but I'll try again. Is it possible that `target_include_directories` with `piof` as an item would work? I'll try that, too.

Reply:
Uhm, that should not work, I think. Only target_link_libraries does the extra magic. The other `target_xyz` do the thing that the name say.

I will check to see if the imported piof target was created correctly in EAMxx. It might be I'm not setting it up correctly.

Reply:
ARG, I copy-pasted stuff for pioc into piof...

```
    add_library(piof INTERFACE)
    target_link_libraries (pioc INTERFACE ${SCORPIO_F_LIB} ${netcdf_f_lib} pioc)
    target_include_directories (pioc INTERFACE ${SCORPIO_INC_DIR} ${netcdf_f_incdir} )
```
Notice the pioc instead of piof...

Reply:
Ok, but just so I'm sure we're on the same page: The goal herre is to get `-I../netcdf-fortran/include` added to the mesh_mod.F90.o target compile line, and nothing more. So, naively, it almost seems like `target_include_directories` ought to do it (except I'm now vaguely recalling it wasn't).

Reply:
I see. I didn't pay enough attention to the AD target lines to catch that bug. Probably explains why I couldn't get the `piof` dependency to do anything.

Reply:
I tested target_link_libraries with the AD fix to piof, and it works. I'll update this PR with that approach.